packages/w3up-client/src/account.js

```diff
@@ -235,6 +235,46 @@ export class AccountPlan {
     })
   }
 
+  /**
+   * Waits for a payment plan to be selected.
+   * This method continuously checks the account's payment plan status
+   * at a specified interval until a valid plan is selected, or when the timeout is reached,
+   * or when the abort signal is aborted.
+   *
+   * @param {object} [options]
+   * @param {number} [options.interval=1000] - The polling interval in milliseconds (default is 1000ms).
+   * @param {number} [options.timeout=300000] - The maximum time to wait in milliseconds before throwing a timeout error (default is 5 minutes).
```

Comment:
I think this is a bit short. Login will timeout after 15 minutes (i.e. it's not possible to login and select a plan after 15 minutes) so it might be an idea to align with that.

https://github.com/storacha/w3up/blob/ea02adb77cfe207a5823e24d886d28b6650ecc03/packages/upload-api/src/access/authorize.js#L41-L42